src/ray/raylet/node_manager.cc

```diff
@@ -2069,17 +2074,19 @@ void NodeManager::ForwardTaskOrResubmit(const Task &task,
       // the task in a little bit. TODO(rkn): Really this should be a
       // unique_ptr instead of a shared_ptr. However, it's a little harder to
       // move unique_ptrs into lambdas.
+      local_queues_.QueueTasks({task}, TaskState::SWAP);
       auto retry_timer = std::make_shared<boost::asio::deadline_timer>(io_service_);
       auto retry_duration = boost::posix_time::milliseconds(
           RayConfig::instance().node_manager_forward_task_retry_timeout_milliseconds());
       retry_timer->expires_from_now(retry_duration);
       retry_timer->async_wait(
-          [this, task, task_id, retry_timer](const boost::system::error_code &error) {
+          [this, task_id, retry_timer](const boost::system::error_code &error) {
             // Timer killing will receive the boost::asio::error::operation_aborted,
             // we only handle the timeout event.
             RAY_CHECK(!error);
             RAY_LOG(INFO) << "Resubmitting task " << task_id
                           << " because ForwardTask failed.";
+            const auto task = local_queues_.RemoveTask(task_id);
```

Comment:
Can you add a comment that the task must be removed from the SWAP state? Is there a way to add an assertion about that?

Reply:
> Nice fix! Is there a good way to test this or trigger this condition?

It's a little difficult to trigger. I was able to get it to fail on `test_component_failures.py::test_raylet_failed` maybe 1/100 times. By decreasing `initial_reconstruction_timeout_milliseconds` to 20ms, I was able to get it to fail sooner.